src/chef-server-ctl/plugins/maintenance.rb

```diff
@@ -0,0 +1,76 @@
+
+require 'redis'
+require "resolv"
+
+def redis
+  @redis ||= begin
+               vip = running_config["private_chef"]["redis_lb"]["vip"]
+               port = running_config["private_chef"]["redis_lb"]["port"]
+               password = credentials.get('redis_lb', 'password')
+               Redis.new(port: port, ip: vip, password: password)
+             end
+end
+
+def enable_maintenance
+  puts "- Enabling maintenance mode"
+  redis.hset("dl_default", "503_mode", true)
+end
+
+def disable_maintenance
+  puts "- Disabling maintenance mode"
+  redis.hdel("dl_default", "503_mode")
+end
+
+def add_ip_to_whitelist(ip)
+  validate_ip(ip)
+  puts "- Adding #{ip} to whitelist."
+  redis.sadd("maint_whitelisted_ip", ip)
+end
+
+def remove_ip_from_whitelist(ip)
+  validate_ip(ip)
+  puts "- Removing #{ip} from whitelist."
+  redis.srem("maint_whitelisted_ip", ip)
+end
+
+def validate_ip(ip)
+  isIPv4 = ip =~ Resolv::IPv4::Regex ? true : false
+  isIPv6 = ip =~ Resolv::IPv6::Regex ? true : false
+  unless isIPv4 or isIPv6
+    $stderr.puts "The IP address is invalid, please enter valid IP address"
+    exit 1
+  end
+end
+
+
+add_command_under_category "maintenance", "general", "Handel the server properly while maintenance", 2 do
+  args = ARGV[1..-1] # Chop off first 1 args, keep the rest... that is, everything after "chef-server-ctl maintenance"
+  options = {}
+
+  if args[0] == 'on'
+    enable_maintenance
+    args = args[1..-1]
+  elsif args[0] == 'off'
+    disable_maintenance
+    args = args[1..-1]
+  elsif args.length == 0
+    $stderr.puts "Please specify the action 'on' or 'off' or pass the maintenance options"
+    exit 1
+  end
```

Comment:
Try a case statement and see if you like it more.  Also, if you put it in a function (method) you could write a unit test for it.

A demo:
```
def validate_args args
	case args[0]
		when 'on'
		'do stuff...'
		'enable maintenance'
		when 'off'
		'do stuff...'
		'disable maintenance'
		else
		'do stuff...'
		"Please specify the action 'on' or 'off' or pass the maintenance options"
	end
end

irb> args = ['on']
irb> validate_args args
=> "enable maintenance"
irb> args = ['off']
=> ["off"]
irb> validate_args
=> "disable maintenance"
irb> args = []
=> []
irb> validate_args
=> "Please specify the action 'on' or 'off' or pass the maintenance options"
irb> args = ['blah']
=> ["blah"]
irb> validate_args
=> "Please specify the action 'on' or 'off' or pass the maintenance options"
```